crypto/txscript/src/lib.rs

```diff
@@ -405,8 +466,13 @@ impl<'a, T: VerifiableTransaction, Reused: SigHashReusedValues> TxScriptEngine<'
         } else if num_sigs > num_keys {
             return Err(TxScriptError::InvalidSignatureCount(format!("more signatures than pubkeys {num_sigs} > {num_keys}")));
         }
+        if let Some(RuntimeSigOpCounter { sig_op_remaining, sig_op_limit }) = &mut self.runtime_sig_op_counting {
+            let num_sigs =
+                num_sigs.try_into().map_err(|_| TxScriptError::NumberTooBig("Signatures count mustn't exceed 255".to_string()))?;
+            *sig_op_remaining =
+                sig_op_remaining.checked_sub(num_sigs).ok_or(TxScriptError::ExceededSigOpLimit(num_sigs as u16, *sig_op_limit))?;
+        }
```

Comment:
I think it would improve readability if `RuntimeSigOpCounter` would have a method that does all of this

Reply:
78dfebd72d7e6a3c0bc0ac3bf410caf9218f7970